Crystalball/crystalball.py

```diff
@@ -160,13 +112,24 @@ def predict():
     # Parse application args
     args = create_parser().parse_args([a for a in sys.argv[1:]])
 
-    with ExitStack() as stack:
-        # Set up dask ThreadPool prior to any application dask calls
-        if args.num_workers:
-            stack.enter_context(dask.config.set(num_workers=args.num_workers))
+    # with ExitStack() as stack:
+    #     # Create a distributed LocalCluster
+    #     cluster = LocalCluster(processes=False,
+    #                            n_workers=args.num_workers,
+    #                            threads_per_worker=1,
+    #                            memory_limit=None)
+    #     # Create distributed client
+    #     client = Client(cluster)
+
+    kwargs = dict(processes=False,
+                  n_workers=args.num_workers,
+                  threads_per_worker=1,
+                  memory_limit='8gb')
```

Comment:
@paoloserra Sorry the memory_limit is specified here.

Reply:
that's cool, I had seen that (and am not changing it for now)

Reply:
This should probably be `memory_fraction*total_memory`, *in this case*.

This is quite configurable and I think the definition of memory_limit changes when each worker is a process (in the above this is not the case `processes=False`).

In any case, I'll be intrigued to see if this runs through.

Reply:
There are a couple of options that can be set here. I'm experimenting with the following:

```python
    system_mem = float(psutil.virtual_memory()[0])
    memory_limit = int(args.memory_fraction*system_mem*2)
    memory_limit = int(0.5*system_mem)
    print("System Memory", system_mem, "Memory Limit", memory_limit)

    kwargs = dict(processes=False,
                  n_workers=args.num_workers,
                  threads_per_worker=1,
                  # Target fraction to stay below
                  memory_target_fraction=False,
                  # Don't spill to disk
                  memory_spill_fraction=False,
                  # Pause workers at 70% of memory_limit
                  memory_pause_fraction=0.7,
                  memory_limit=memory_limit
```

but can't get it to behave stably at the moment.